src/backend/catalog/namespace.c

```diff
@@ -1495,18 +1495,34 @@ LookupInternalNamespaceId(const char *nspname)
 Oid
 LookupNamespaceId(const char *nspname, Oid dboid)
 {
-    if(gp_upgrade_mode){
+
+	/* check for pg_temp alias */
+	if (NSPDBOID_CURRENT == dboid && strcmp(nspname, "pg_temp") == 0)
+	{
+		if (TempNamespaceValid(true))
+			return myTempNamespace;
+		/*
+		 * Since this is used only for looking up existing objects, there
+		 * is no point in trying to initialize the temp namespace here;
+		 * and doing so might create problems for some callers.
+		 * Just fall through and give the "does not exist" error.
+		 */
+	}
+
+    if(gp_upgrade_mode)
```

Comment:
I am thinking we may be able to remove the entire if(gp_upgrade_mode), as there's no upgrade path from previous Pivotal commercial 1.3.x HAWQ to 2.0 (only way is to do data reloading).
So this logic probably is no longer needed.